templates/linux/squeak.sh

```diff
@@ -4,14 +4,54 @@
 # Version:     2.4
 # Date:        2020/03/31
 # Description: Script to run Squeak from the Linux bundle
+# usage:
+#    squeak [<vmoptions>] *.image [ <stscript> [ <st args> ... ]]
+
+APP_NAME=Squeak
 
 # paths
 DIR=$(readlink -f "$0") #resolve symlink
 ROOT=$(dirname "${DIR}") #obtain dir of the resolved path
-VM="${ROOT}/bin/squeak"
-RESOURCES="${ROOT}/shared"
-CONF_FILE="/etc/security/limits.d/squeak.conf"
-IMAGE=$*
+CONF_FILE="/etc/security/limits.d/${APP}.conf"
+OS=$(uname -s)
+CPU=$(uname -m)
+case "${CPU}" in
+    "x86_64") CPU="i686" ;;
+    "armv6l"|"armv7l") CPU="ARM" ;;
+esac
+
+if [ -d ${ROOT}/bin ]; then
+    BINDIR="${ROOT}/bin"
+    RESOURCES="${ROOT}/shared"
+else
+    # all-in-one bundle
+    appdir=$(echo ${APP_NAME}*.app/)
+    BINDIR="${appdir}/Contents/${OS}-${CPU}/"
+    RESOURCES="${appdir}/Contents/Resources/"
+fi
+
+APP=$(echo $APP_NAME | tr [:upper:] [:lower:])
```

Comment:
```suggestion
APP=$(echo "${APP_NAME}" | tr [:upper:] [:lower:])
```

Reply:
APP and APP_NAME are extracted from script basename (root) part. So it should work without  changes for squeak/Squeak, cuis/Cuis, pharo/Pharo, newspeak/Newspeak etc.